Core/src/Core/Filters/CalculateArrayHistogramFilter.cpp

```diff
@@ -0,0 +1,155 @@
+#include "CalculateArrayHistogramFilter.hpp"
+#include "Algorithms/CalculateArrayHistogram.hpp"
+
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataPath.hpp"
+#include "complex/DataStructure/IDataArray.hpp"
+#include "complex/Filter/Actions/CreateArrayAction.hpp"
+#include "complex/Filter/Actions/CreateDataGroupAction.hpp"
+#include "complex/Parameters/ArrayCreationParameter.hpp"
+#include "complex/Parameters/ArraySelectionParameter.hpp"
+#include "complex/Parameters/BoolParameter.hpp"
+#include "complex/Parameters/DataGroupCreationParameter.hpp"
+#include "complex/Parameters/DataGroupSelectionParameter.hpp"
+#include "complex/Parameters/MultiArraySelectionParameter.hpp"
+#include "complex/Parameters/NumberParameter.hpp"
+
+using namespace complex;
+
+namespace complex
+{
+//------------------------------------------------------------------------------
+std::string CalculateArrayHistogramFilter::name() const
+{
+  return FilterTraits<CalculateArrayHistogramFilter>::name.str();
+}
+
+//------------------------------------------------------------------------------
+std::string CalculateArrayHistogramFilter::className() const
+{
+  return FilterTraits<CalculateArrayHistogramFilter>::className;
+}
+
+//------------------------------------------------------------------------------
+Uuid CalculateArrayHistogramFilter::uuid() const
+{
+  return FilterTraits<CalculateArrayHistogramFilter>::uuid;
+}
+
+//------------------------------------------------------------------------------
+std::string CalculateArrayHistogramFilter::humanName() const
+{
+  return "Calculate Frequency Histogram";
+}
+
+//------------------------------------------------------------------------------
+std::vector<std::string> CalculateArrayHistogramFilter::defaultTags() const
+{
+  return {"#Statistics", "#Ensemble"};
+}
+
+//------------------------------------------------------------------------------
+Parameters CalculateArrayHistogramFilter::parameters() const
+{
+  Parameters params;
+  // Create the parameter descriptors that are needed for this filter
```

Comment:
try to organize the parameters a bit by adding in a few:
```
  params.insertSeparator(Parameters::Separator{"FILL IN SOME TITLE HERE"});
```
I try to group things as:
* Input Parameters
* Required Data Objects
* Created Data Objects